src/eip7702/ZeroExEIP7702Wallet.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {IERC20} from "@forge-std/interfaces/IERC20.sol";
+import {IERC5267} from "../interfaces/IERC5267.sol";
+import {ISettlerBase} from "../interfaces/ISettlerBase.sol";
+import {ISettlerTakerSubmitted} from "../interfaces/ISettlerTakerSubmitted.sol";
+
+import {Context} from "../Context.sol";
+import {SettlerSwapper} from "../SettlerSwapper.sol";
+
+import {SafeTransferLib} from "../vendor/SafeTransferLib.sol";
+import {FastLogic} from "../utils/FastLogic.sol";
+import {LibZip} from "../vendor/LibZip.sol";
+
+contract ZeroExEIP7702Wallet is IERC5267, Context, SettlerSwapper {
+    using FastLogic for bool;
+    using SafeTransferLib for IERC20;
+
+    address private immutable _cachedThis;
+    string public constant name = "ZeroExEIP7702Wallet";
+    uint256 private immutable _cachedChainId;
+    bytes32 private immutable _cachedDomainSeparator;
+
+    error DeploymentFailed();
+    error PermissionDenied();
+
+    modifier onlyWallet() {
+        require(address(this).code.length == 23);
```

Comment:
```suggestion
        require(address(this).code.length == 38);
```

Reply:
This is supposed to run in 7702'd EOAs that have exactly 23 bytes of code, beginning with 0xef0100